common/src/main/java/io/smallrye/config/common/utils/StringUtil.java

```diff
@@ -171,9 +171,8 @@ public static String toLowerCaseAndDotted(final String name) {
 
                 // Do not convert to index if the first segment is a number
                 if (beginSegment > 0) {
-                    try {
-                        String segment = sb.substring(beginSegment, i);
-                        Integer.parseInt(segment);
+                    String segment = sb.substring(beginSegment, i);
+                    if (isNumeric(segment)) {
```

Comment:
No need to substring here, just expose a ranged check...and performing the parsing on it @radcortez

Reply:
Sure.